ufm_pkeys/ufm_pkeys.py

```diff
@@ -0,0 +1,224 @@
+        },
+        {
+            "name": f'--{PKEYS_OPERATIONS.get("delete_pkey")}',
+            "help": "Option to delete specific Pkey"
+        }
+    ]
+
+
+
+class UfmPkeysConfigParser(ConfigParser):
+    def __init__(self, args):
+        super().__init__(args)
+
+
+
+
+class UfmPkeysManagement:
+
+    @staticmethod
+    def set_pkey(pkey,ports_guids_list,memberships= None,
+                    default_membership = "full", index0 = False, ip_over_ib = True):
+        payload = {
+            UfmPkeysConstants.API_GUIDS:ports_guids_list,
+            UfmPkeysConstants.API_PKEY: pkey,
+            UfmPkeysConstants.API_INDEX0 : index0,
+            UfmPkeysConstants.API_IP_OVER_IB : ip_over_ib
+        }
+
+        if memberships and len(memberships) :
+            payload[UfmPkeysConstants.API_MEMBERSHIPS] = memberships
+        else:
+            payload[UfmPkeysConstants.API_MEMBERSHIP] = default_membership
+
+        response = ufm_rest_client.send_request(UfmPkeysConstants.PKEYS_API_URL, HTTPMethods.PUT, payload=payload)
+        if response and response.status_code == HTTPStatus.OK:
+            Logger.log_message(f'The pkey: {pkey} has been set successfully')
+        else:
+            Logger.log_message(f'The pkey: {pkey} hasn`t been set', LOG_LEVELS.ERROR)
+        return response
+
+    @staticmethod
+    def get_pkeys(pkey=None):
+        if pkey:
+            url = f'{UfmPkeysConstants.PKEYS_API_URL}/{pkey}'
+        else:
+            url = UfmPkeysConstants.PKEYS_API_URL
+        url = f'{url}?guids_data=true'
+        response = ufm_rest_client.send_request(url)
+        Logger.log_message(response.json())
+        return response
+
+    @staticmethod
+    def delete_pkey(pkey):
+        url = f'{UfmPkeysConstants.PKEYS_API_URL}/{pkey}'
+        response = ufm_rest_client.send_request(url, HTTPMethods.DELETE)
+        if response:
+            if response.status_code == HTTPStatus.OK:
+                Logger.log_message(f'The Pkey: {pkey} has been removed successfully')
+            elif response.status_code == HTTPStatus.NOT_FOUND:
+                Logger.log_message(f'The Pkey: {pkey} is not found', LOG_LEVELS.ERROR)
+        else:
+            Logger.log_message(f'The Pkey: {pkey} hasn`t been removed successfully', LOG_LEVELS.ERROR)
+        return response
+
+
+
+
+
+if __name__ == "__main__":
+    # init app args
+    args = ArgsParser.parse_args("UFM Pkeys Management", UfmPkeysConstants.args_list)
+
+    # init app config parser & load config files
+    config_parser = UfmPkeysConfigParser(args)
+
+    # init logs configs
+    logs_file_name = config_parser.get_logs_file_name()
+    logs_level = config_parser.get_logs_level()
+    Logger.init_logs_config(logs_file_name, logs_level)
+
+
+    # init ufm rest client
+    ufm_rest_client = UfmRestClient(host = config_parser.get_ufm_host(),
+                                    username=config_parser.get_ufm_username(),
```

Comment:
what about ufm token? it should be the default, user/pw will be deprecated at some point

Reply:
handled in the below commit
https://github.com/Mellanox-lab/ufm_sdk/pull/28/commits/658810598f7eaf1e6831ea2005c9b674fc816f48